Sources/Flare/Classes/Providers/PaymentProvider/PaymentProvider.swift

```diff
@@ -125,12 +125,20 @@ extension PaymentProvider: SKPaymentTransactionObserver {
                     forKey: transaction.payment.productIdentifier
                 ), !handlers.isEmpty {
                     self.dispatchQueueFactory.main().async {
-                        handlers.forEach { $0(queue, .success(transaction)) }
+                        if let error = transaction.error {
```

Comment:
This is only true if the transaction state is set to `.failed`. A user can manually check the `error` and `transactionState` properties of `PaymentTransaction`, which should have these values.

I'm not sure we need to throw an error here.

Reply:
Sometimes it is necessary to collect error information to improve the payment experience.

Reply:
@zy19890102, why can't you collect the error information using `PaymentTransaction`? Could you please provide a minimal example where these changes are necessary?

Reply:
When users cancel payments, we will ask them for the reason behind the cancellation, whether it is due to the pricing of the product being too high or the subscription service not meeting their needs, in order to adjust our pricing strategy and subscription benefits.

Reply:
@zy19890102 I see, thank you. But I don't get why you can't use the `error` property from the transaction.

There is a merge conflict. Please rebase, and could you also run the make fmt command in your terminal to format the codebase according to the project’s code style?